### ISSUE TITLE
Increase general library quality

### ISSUE THREAD

**Issue by MatthijsBurgh** [RobotWebTools/roslibjs#647]
The [robostack fork](https://github.com/RoboStack/roslibjs), has applied some improvements like:
- Migrating to typescript
- Migrating to ES6 classes
- Replacing grunt by rollup

This might be good improvements to apply to the original library as well. Preferably we include these changes also in the `ros3djs` and `ros2djs` libraries.

**Comment by EzraBrooks** [comment]
Agreed. I would recommend we start by adding a TypeScript compiler configuration that allows JavaScript files, fixing any issues with statically type checking the current JS, then move forward with TS.

**Comment by EzraBrooks** [comment]
Once #665 (and all the dependent PRs that have come out of it) are in, I plan on making a PR to switch from Grunt + Browserify + Karma to Vite (which includes Rollup) and Vitest, as that's what I have experience with using lately and I think Vite's "library mode" is a very sensible preset for a package like roslibjs.
